hack/sync-manifests.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This shell script is used to automatically syncing the Kustomize manifests from manifests templated by trainer Helm chart.
+# Run 'make sync-manifests' from the root directory of this repository.
+
+echo "Syncing Kustomize manifests from manifests templated by trainer Helm chart..."
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+# Helm chart release name and namespace.
+TRAINER_CHART_DIR=charts/kubeflow-trainer
+RELEASE_NAME=kubeflow-trainer
+RELEASE_NAMESPACE=kubeflow-system
+
+# Source directories
+SRC_CRD_DIR=manifests/base/crds
+
+# Destination directories
+DST_CRD_DIR=charts/trainer/crds
+DST_RBAC_DIR=manifests/base/rbac
+DST_CONTROLLER_DIR=manifests/base/controller
+DST_WEBHOOK_DIR=manifests/base/webhook
+DST_RUNTIMES_DIR=manifests/base/runtimes
+
+MANIFESTS_FILE=$(mktemp -t kubeflow-trainer-manifests.yaml)
+FIND_EXCLUDE_ARGS="-not -name kustomization.yaml -not -name kustomization.yml"
+YEAR=$(date +%Y)
+LICENSE=$(cat hack/boilerplate/boilerplate.go.txt | sed 's|//|#|' | sed "s|YEAR|${YEAR}|g")
+YQ_ARGS=".metadata.labels.\"app.kubernetes.io/managed-by\" = \"Kustomize\" | del(.metadata.labels.\"helm.sh/chart\") | ... comments=\"\""
+
+setup() {
+    # yq is required to parse yaml files.
+    if ! command -v yq &>/dev/null; then
+        echo "'yq' is not installed, please install it first. Ref: https://github.com/mikefarah/yq."
+        exit 1
+    fi
+
+    # Create destination directory if it doesn't exist.
+    mkdir -p ${DST_CRD_DIR} ${DST_RBAC_DIR} ${DST_CONTROLLER_DIR} ${DST_WEBHOOK_DIR} ${DST_RUNTIMES_DIR}/pretraining
+
+    helm template ${RELEASE_NAME} ${TRAINER_CHART_DIR} --namespace ${RELEASE_NAMESPACE} > "${MANIFESTS_FILE}"
+}
+
+update_crds() {
+    # Copy all CRD files to destination directory.
+    # shellcheck disable=SC2086
+    find ${SRC_CRD_DIR} -type f -name "*.yaml" ${FIND_EXCLUDE_ARGS} -exec cp {} ${DST_CRD_DIR} \;
+}
+
+update_rbac() {
+    printf '#\n%s\n#\n\n' "${LICENSE}" > ${DST_RBAC_DIR}/serviceaccount.yaml
+    yq -e "select(.kind == \"ServiceAccount\" and .metadata.name == \"kubeflow-trainer-controller\") | ${YQ_ARGS}" "${MANIFESTS_FILE}" >> ${DST_RBAC_DIR}/serviceaccount.yaml
```

Comment:
I guess, the name should be updated to `kubeflow-trainer-controller-manager`:
https://github.com/kubeflow/trainer/blob/master/manifests/base/manager/manager.yaml#L5